sw-emulator/lib/periph/src/soc_reg.rs

```diff
@@ -820,11 +820,15 @@ impl SocRegistersImpl {
         if size != RvSize::Word {
             Err(BusError::StoreAccessFault)?
         }
+        self.bus_warm_reset();
```

Comment:
I don't think you need top call this. bus_warm_reset() should be invoked after you send TimerAction::Warmreset.

Reply:
just one question, how does the bus_warm_reset gets called?